color/src/colorspace.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright 2024 the Color Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use core::f32;
+
+use crate::{matmul, tagged::ColorspaceTag};
+
+#[cfg(all(not(feature = "std"), not(test)))]
+use crate::floatfuncs::FloatFuncs;
+
+/// The main trait for colorspaces.
+///
+/// This can be implemented by clients for conversions in and out of
+/// new colorspaces. It is expected to be a zero-sized type.
+///
+/// The linear sRGB colorspace is central, and other colorspaces are
+/// defined as conversions in and out of that. A colorspace does not
+/// explicitly define a gamut, so generally conversions will succeed
+/// and round-trip, subject to numerical precision.
+///
+/// White point is not explicitly represented. For color spaces with a
+/// white point other than D65 (the native white point for sRGB), use
+/// a linear Bradford chromatic adaptation, following CSS Color 4.
+pub trait Colorspace: Clone + Copy + 'static {
```

Comment:
I'd *probably* expect this to be `ColorSpace`; the css color spec uses the term "colorspace" 9 times, and the term "color space" ~200 times (of note, it also uses the term `color-space` ~9 times)